servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -612,19 +633,29 @@ private TraversalResult traverseToInode(String[] pathComponents, boolean collect
      */
     private final List<Inode> mNonPersisted;
 
-    static TraversalResult createFoundResult(Inode inode, List<Inode> nonPersisted) {
-      return new TraversalResult(true, -1, inode, nonPersisted);
+    /**
+     * The list of all inodes encountered during the traversal, including null elements in the
+     * list if corresponding inodes in the path are not found.
+     */
+    private final List<Inode> mInodesIncludeNull;
```

Comment:
this should be just inodes and it should store the inodes encountered during the traversal